src/main/java/com/google/gcs/sdrs/service/worker/impl/ExecuteRetentionWorker.java

```diff
@@ -62,6 +63,53 @@ public void doWork() {
     String dataStorageName = getDataStorageName(executionEvent.getTarget());
     String projectId = executionEvent.getProjectId();
     RetentionRule rule;
+    try {
+      switch (executionEvent.getExecutionEventType()) {
+        case USER_COMMANDED:
+          List<RetentionRule> userRules = new ArrayList<>();
+          userRules.add(getEventDefinedRule());
+          executeRules(userRules, projectId);
+          break;
+        case POLICY:
+          boolean dataStorageExists = !dataStorageName.isEmpty();
+          boolean projectIdExists = projectId != null && !projectId.isEmpty();
+
+          if (projectIdExists && dataStorageExists) {
+            rule = retentionRuleDao.findDatasetRuleByBusinessKey(projectId, dataStorageName);
+            if (rule != null) {
+              List<RetentionRule> datasetRules = new ArrayList<>();
+              datasetRules.add(getEventDefinedRule());
+              executeRules(datasetRules, projectId);
+            } else {
+              String message =
+                  String.format(
+                      "No policy found for target project: %s, target: %s",
+                      projectId, dataStorageName);
+              logger.error(message);
+              throw new UnsupportedOperationException(message);
+            }
+
+          } else if (projectIdExists) {
+            executeAllDatasetRulesByProject(executionEvent.getProjectId());
+          } else {
+            executeAllDatasetRules();
+          }
+          break;
+        default:
+          logger.warn(
+              String.format(
+                  "%s is not a supported retention execution type",
+                  executionEvent.getExecutionEventType().toString()));
+      }
+      workerResult.setStatus(WorkerResult.WorkerResultStatus.SUCCESS);
+    } catch (IOException | IllegalArgumentException ex) {
+      logger.error(String.format("Error executing rule: %s", ex.getMessage()));
+      workerResult.setStatus(WorkerResult.WorkerResultStatus.FAILED);
+    }
+
+    /*   String dataStorageName = getDataStorageName(executionEvent.getTarget());
+    String projectId = executionEvent.getProjectId();
```

Comment:
Remove commented out code if not needed.

Reply:
done